### ISSUE TITLE
Deployment

### ISSUE THREAD

**Issue by armanakbari** [Robbyant/lingbot-va#85]
Thanks for the great work. You have propose the FDM-grounded strategy for deployment in real world. It seems that you have included that in the released code.

I have some questions regarding that. How did you add the forward dynamic loss to the training? do you do another forward pass to calculate that? is the backward pass separate for this and other losses?

**Comment by zqh0253** [comment]
Thanks for your interest. For each sequence, we randomly compute either the FDM or IDM loss

**Comment by armanakbari** [comment]
> Thanks for your interest. For each sequence, we randomly compute either the FDM or IDM loss

thanks for the quick response. what did you set as the coefficient of FDM loss (Lambda)? so for both of them, we always have the l_dyn loss? or we remove the l_dyn for FDM?

**Comment by zqh0253** [comment]
We set 1 as the coefficient and remove dynamic loss for FDM.

**Comment by armanakbari** [comment]
> We set 1 as the coefficient and remove dynamic loss for FDM.

Thanks. For inference:
1. Did your deployment use a single-chunk cold start as written in Algorithm 2? If so, how did you handle the first iteration's grounding/frame-index without re-grounding C0?

2. At deployment, is the FDM imagine pass a single-step (quasi-deterministic forward-dynamics) prediction, or a few-step partial denoise to a target σ? Does it integrate fully to s=1 (clean) or stop at a partial σ like the action-conditioning path?

**Comment by zqh0253** [comment]
1. Yes. Grounding is not required for cold-starting the first chunk. 
2. We perform multi-step denoising and integrate to the clean timestep, though there is room for improvement by switching to partial denoising.

**Comment by armanakbari** [comment]
> 1. Yes. Grounding is not required for cold-starting the first chunk.
> 2. We perform multi-step denoising and integrate to the clean timestep, though there is room for improvement by switching to partial denoising.

Thanks for the response. I implemeted this and it is working well when I fine-tune on one task. But when I fine-tune on 5 different tasks the loss curves does not decrease that much which leads to success rate reduction. 

This is the loss curve for fine tuning on 5 tasks: 
<img width="1453" height="687" alt="Image" src="https://github.com/user-attachments/assets/072d08bb-411c-43fb-8ff2-9734b7269539" />

and this is the loss curve for one task:

<img width="1442" height="661" alt="Image" src="https://github.com/user-attachments/assets/e90e49a5-54c1-41af-9356-3ae09b74c1e1" />

I would appriciate if you help me with the following questions:

1. should I train for more steps for 5 tasks?
2. do you fine tune on all of your real-world tasks or you have different individual checkpoints for each task?
